otelcol/collector.go

```diff
@@ -15,6 +15,7 @@ import (
 
 	"go.uber.org/multierr"
 	"go.uber.org/zap"
+	"go.uber.org/zap/zaptest/observer"
```

Comment:
We're certainly using the package in a use case I think they didn't intend. I still think the simplicity of https://github.com/open-telemetry/opentelemetry-collector/pull/10007 is most appropriate as it is the collector making logging decisions before being told how to log.  It keeps the logs in proper order as well

Reply:
> I worry that this package may change in the future causing issue, but maybe it's not a problem?

I don't think that should be a problem since (i) the fact that we use this package is an implementation detail and (ii) even if the package disappeared we could rebuild it ourselves (it does not depend on any internal bits of zap)